### PR TITLE
Add random id to rendered div to support multiple open-components of …

### PR DESCRIPTION
**Pull request by kenhkelly**
…the same type, on the same page.

### CODE REVIEW

File: acceptance-setup/__tests__/acceptance.js [hunk 1]
```diff
@@ -80,6 +80,7 @@ test("Registry should correctly serve rendered and unrendered components", done
       const bodyVersionless = body
         .replace(semverRegex, "6.6.6")
         .replace(/data-hash=\\\".*?\\\"/, "")
+        .replace(/oc-reactRoot-react-app-.*?\"/g, 'oc-reactRoot-react-app-1.1"')
```
Comment (matteofigus) [inline acceptance-setup/__tests__/acceptance.js]: I think this improves readability?

```suggestion
        .replace(/oc-reactRoot-react-app-.*?\"/g, 'oc-reactRoot-react-app-randomId"')
```

File: acceptance-setup/__tests__/acceptance.js [hunk 2]
```diff
@@ -113,6 +114,7 @@ test("server-side-side rendering", done => {
       const domVersionless = dom
         .serialize()
         .replace(semverRegex, "6.6.6")
+        .replace(/oc-reactRoot-react-app-.*?\"/g, 'oc-reactRoot-react-app-1.1"')
```
Comment (matteofigus) [inline acceptance-setup/__tests__/acceptance.js]:
```suggestion
        .replace(/oc-reactRoot-react-app-.*?\"/g, 'oc-reactRoot-react-app-randomId"')
```

File: packages/oc-template-react-compiler/lib/viewTemplate.js [hunk 3]
```diff
@@ -8,7 +8,9 @@ const viewTemplate = ({
   var modelHTML =  model.__html ? model.__html : '';
   var staticPath = model.reactComponent.props._staticPath;
   var props = JSON.stringify(model.reactComponent.props);
-  return '<div id="${reactRoot}" class="${reactRoot}">' + modelHTML + '</div>' +
+  var randomId = Math.random() * 10000000;
+  var reactRoot = "${reactRoot}" + '-' + randomId;
```
Comment (matteofigus) [inline packages/oc-template-react-compiler/lib/viewTemplate.js]: If we just override the reactRoot variable, we change both the `id` and `class` attribute in the markup. I don't think we need to change the class too as that could break some stuff. I propose we change only the id: 

```suggestion
  var reactRootId = "${reactRoot}-" + randomId;
```

File: packages/oc-template-react-compiler/lib/viewTemplate.js [hunk 4]
```diff
@@ -8,7 +8,9 @@ const viewTemplate = ({
...
+  return '<div id="'+ reactRoot +'" class="'+ reactRoot +'">' + modelHTML + '</div>' +
```
Comment (matteofigus) [inline packages/oc-template-react-compiler/lib/viewTemplate.js]: ^ follows previous suggestion
```suggestion
  return '<div id="'+ reactRootId +'" class="${reactRoot}">' + modelHTML + '</div>' +
```

File: packages/oc-template-react-compiler/lib/viewTemplate.js [hunk 5]
```diff
@@ -20,7 +22,7 @@ const viewTemplate = ({
           '["oc", "reactComponents", "${bundleHash}"],' + 
           '"' + staticPath + '${bundleName}.js",' +
           'function(ReactComponent){' +
-            'var targetNode = document.getElementById("${reactRoot}");' +
+            'var targetNode = document.getElementById("'+ reactRoot +'");' +
```
Comment (matteofigus) [inline packages/oc-template-react-compiler/lib/viewTemplate.js]: ^ Follows previous suggestions:
```suggestion
            'var targetNode = document.getElementById("'+ reactRootId +'");' +
```

**Comment by kenhkelly** [comment]
@matteofigus Thanks for the suggestions, I have incorporated them.

**Pull request merged**